sdk/identity/identity/src/credentials/defaultAzureCredential.ts

```diff
@@ -233,17 +233,49 @@ export class DefaultAzureCredential extends ChainedTokenCredential {
   constructor(options?: DefaultAzureCredentialOptions);
 
   constructor(options?: DefaultAzureCredentialOptions) {
-    const credentialFunctions = [
-      createEnvironmentCredential,
-      createDefaultWorkloadIdentityCredential,
-      createDefaultManagedIdentityCredential,
+    // If AZURE_TOKEN_CREDENTIALS is not set, use the default credential chain.
+    const azureTokenCredentials = (process.env.AZURE_TOKEN_CREDENTIALS)?.toLowerCase();
+    let devCredentialFunctions = [
       createDefaultAzureCliCredential,
       createDefaultAzurePowershellCredential,
       createDefaultAzureDeveloperCliCredential,
     ];
+    let prodCredentialFunctions = [
+      createEnvironmentCredential,
+      createDefaultWorkloadIdentityCredential,
+      createDefaultManagedIdentityCredential,
+    ];
+    let credentialFunctions = [];
+    // If AZURE_TOKEN_CREDENTIALS is set, use it to determine which credentials to use.
+    // The value of AZURE_TOKEN_CREDENTIALS should be either "dev" or "prod".
+    if (azureTokenCredentials) {
```

Comment:
Trim this string before running the string comparison in the switch statement below

Reply:
Thanks! added that before assigning.